File: common/src/test/java/dev/number6/generate/DetectSentimentResultGenerator.java
package dev.number6.generate;


import com.amazonaws.services.comprehend.model.DetectSentimentResult;
import com.amazonaws.services.comprehend.model.SentimentType;
import uk.org.fyodor.generators.Generator;

public class DetectSentimentResultGenerator implements Generator<DetectSentimentResult> {

    SentimentScoreGenerator sentimentScoreGenerator = new SentimentScoreGenerator();
    Generator<SentimentType> sentimentTypeGenerator = CommonRDG.value(SentimentType.class);

    @Override
    public DetectSentimentResult next() {
        return new DetectSentimentResult()
                .withSentiment(sentimentTypeGenerator.next())
                .withSentimentScore(sentimentScoreGenerator.next());
    }

}

File: common/src/test/java/dev/number6/generate/DetectSentimentResultGenerator.kt
package dev.number6.generate

import com.amazonaws.services.comprehend.model.DetectSentimentResult
import com.amazonaws.services.comprehend.model.SentimentType
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG

class DetectSentimentResultGenerator : Generator<DetectSentimentResult> {
    var sentimentScoreGenerator = SentimentScoreGenerator()
    var sentimentTypeGenerator = RDG.value(SentimentType::class.java)
    override fun next(): DetectSentimentResult {
        return DetectSentimentResult()
                .withSentiment(sentimentTypeGenerator.next())
                .withSentimentScore(sentimentScoreGenerator.next())
    }
}